### PR TITLE
Display extrinsic meta documentation on hover

### LINKED ISSUE #969: Display extrinsic meta documentation on hover
Goal:

To make specific extrinsic information available at the fingertips of the user, as opposed to referencing more external documentation.

Background:

Currently the metadata returns in-code documentation that expands on the methods for both the extrinsics and chain state. This information is used to display a very condensed summary in the extrinsic selection, i.e.

image

We wish to expand this to actually show all information.

Actions:

- Expand the `submit the following extrinsic` label (or any label passed for an Extrinsic/Proposal to contain `help` with the actual documentation.
- As the method selection changes, the updated documentation becomes available as part of the hover. (This is actually better suited to the dropdown itself, and _may_ work as a tooltip - but these things are large)

Bonus:

Also apply to the chain state app

### PR DESCRIPTION
**Pull request by pbteja1998**
Resolves #969 

---

### Extrinsic

Screenshot from 2019-05-31 00-31-37 (1) (1)

Screenshot from 2019-05-31 14-22-56 (1)

---

### Chain State
Screenshot from 2019-05-31 00-40-49 (1)

---

### CODE REVIEW

**Comment by pbteja1998** [comment]
@jacogr Other than these two, any thing else needs to be changed?

File: packages/app-storage/src/Selection/Modules.tsx [hunk 1]
```diff
@@ -55,6 +55,11 @@ class Modules extends TxComponent<Props, State> {
             defaultValue={this.defaultValue}
             label={t('selected state query')}
             onChange={this.onChangeKey}
+            help={
+              t(meta && meta.documentation
```
Comment (jacogr) [inline packages/app-storage/src/Selection/Modules.tsx]: We don't need to add the `t` wrapper around here (these come directly from the node, so is not translatable), and would suggest simply -

`help={meta && meta.documentation && meta.documentation.join(' ')}`

File: packages/ui-app/src/Extrinsic.tsx [hunk 2]
```diff
@@ -61,6 +61,11 @@ class ExtrinsicDisplay extends React.PureComponent<Props, State> {
           label={label}
           onChange={this.onChangeMethod}
           withLabel={withLabel}
+          help={
+            meta && meta.documentation
```
Comment (jacogr) [inline packages/ui-app/src/Extrinsic.tsx]: As per above comment around just using `&&`

**Comment by pbteja1998** [comment]
@jacogr Made the requested changes.

**Pull request merged**